atlp-rwanda/script-heroes-bn-backend
# #174797516 Email Verification

#### What does this PR do?
Send a user a verification email upon signing up.
#### Description of Task to be completed?
- Setting up sendgrid as an external source for sending emails.
- Configuring sendgrid to send verification email upon successful registration
- Create an end point to verify an email
#### How should this be manually tested?
- Clone this repo:`https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch working directory:`cd script-heroes-bn-backend`
- Checkout to the branch:`git checkout ft-auto-registration-email-#174797516`
- Create a .env file and update it using the example in the env-example file.
-  Install all dependencies: `npm install`
- Run your migrations
- Start your server: `npm run dev`
- Go to postman and sign up
- Upon successful signup you should receive an email within 5 seconds,click on the verify email button to verify your email
- You can then Sign in.

#### What are the relevant pivotal tracker stories?
[#174797516](https://www.pivotaltracker.com/n/projects/2465784)

<img width="1437" alt="Screen Shot 2020-10-06 at 09 19 26" src="https://user-images.githubusercontent.com/51243753/95170843-574bea00-07b5-11eb-84cf-7f1f6ba32120.png">
<img width="1440" alt="Screen Shot 2020-10-06 at 09 19 40" src="https://user-images.githubusercontent.com/51243753/95170871-5fa42500-07b5-11eb-998c-9ea8b2c11197.png">
